src/projection.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    ProjectTo(x)
+
+Returns a `ProjectTo{P,...}` functor able to project a differential `dx` onto the type `P`
+for a primal `x`.
```

Comment:
We need to not use the world primal
```suggestion
Returns a `ProjectTo{T,...}` functor able to project a differential `dx` onto the same tangent space as `x`.
```